usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -1036,6 +1036,8 @@ uint64_t l2arc_feed_min_ms = L2ARC_FEED_MIN_MS;	/* min interval milliseconds */
 boolean_t l2arc_noprefetch = B_TRUE;		/* don't cache prefetch bufs */
 boolean_t l2arc_feed_again = B_TRUE;		/* turbo warmup */
 boolean_t l2arc_norw = B_TRUE;			/* no reads during writes */
+boolean_t arc_evict_l2_first = B_FALSE;		/* first evict buffers from ARC which are in L2ARC */
+boolean_t arc_evict_l2_only = B_FALSE;		/* only evict buffers from ARC which are in L2ARC */
```

Comment:
Does this mean that we don't evict the requested amount from the ARC if it would evict non-L2ARC bufs?  That seems like a bad idea, as it could lead to running out of memory.